joomla/joomla-cms
# [staging] Fix overrides usability

Pull Request for Issue https://github.com/joomla/joomla-cms/issues/22014
Replacing https://github.com/joomla/joomla-cms/pull/22105 (after picking some stuff from it, thanks @wilsonge 

### Summary of Changes
Separate the `language_client` field from searchtools to make it obvious what the user has to do to see the existing overrides.
(adding a bar.php for overrides)
Adding a `Select Language & Client` to the field to never have a default language/client already chosen when creating a new override (this was already extremely confusing for users with the old filter).
Taking off therefore the default site language/client and display a warning if trying to create a new override without choosing one in the dropdown.

### Testing Instructions
Install some languages.
Patch and test that creating overrides works for each language/client selected (files created in the correct overrides folder in admin and site), including when choosing "both" when creating an admin override.
Test that filtering and searching works.
Test that the warning is displayed when clicking on `New` while the language_client selection is set to `Select Language & Client`
Test filtering 

### Before patch

![oldoverride](https://user-images.githubusercontent.com/869724/45288190-5b2fb400-b4ea-11e8-8976-44a08748f091.gif)

### After patch
Basic Manager new look

![screen shot 2018-09-10 at 10 58 03](https://user-images.githubusercontent.com/869724/45287500-b82a6a80-b4e8-11e8-9894-ee3f0c55be2b.png)

Dropdown choice

![screen shot 2018-09-10 at 11 03 08](https://user-images.githubusercontent.com/869724/45287672-21aa7900-b4e9-11e8-90ca-c7e01bf49ad2.png)

When a language/client has been selected (this one contains an override both in site and admin)

![screen shot 2018-09-10 at 11 02 09](https://user-images.githubusercontent.com/869724/45287632-063f6e00-b4e9-11e8-9cb3-5b389eb7fd33.png)

Creating a new override when the field is set to a specific language/client (Here French Administrator)

![screen shot 2018-09-10 at 11 05 27](https://user-images.githubusercontent.com/869724/45287795-74843080-b4e9-11e8-9e17-fa6d63f8919b.png)

Displaying a Warning and redirected to the overrides view when the field is still set to `Select Language & Client`

![overridewarning](https://user-images.githubusercontent.com/869724/45287917-c1680700-b4e9-11e8-9555-dc691a6bdfed.gif)

Suggestions welcome to improve.

@wilsonge @mbabker
